### ISSUE TITLE
Length cannot be stored as unsigned

### ISSUE THREAD

**Issue by gokaco** [typetools/checker-framework#2483]
This error has come repeatedly in case studies where length was used as unsigned but due to weakness in signedness checker, we had to suppress the error.
Command-
```
javac -processor signedness ex.java
```
Code-
```
import org.checkerframework.checker.signedness.qual.*;
class ex{
	void foo(String a,byte []b){
		@Unsigned int len = a.length();
		@Unsigned int len2 = b.length;
	}
}
```
Output-
```
ex.java:4: error: [assignment.type.incompatible] incompatible types in assignment.
		@Unsigned int len = a.length();
		                            ^
  found   : @Signed int
  required: @Unsigned int
ex.java:5: error: [assignment.type.incompatible] incompatible types in assignment.
		@Unsigned int len2 = b.length;
		                      ^
  found   : @Signed int
  required: @Unsigned int
2 errors
`

**Comment by mernst** [comment]
More generally, `@NonNegative int` should be treated as `@Constant` (which will later be renamed to `@SignednessEither`).

**Comment by mernst** [comment]
The Signedness Checker needs to read annotations from some other checker:
 * Index Checker annotations `@NonNegative` (and maybe `@Positive`), or
 * Constant Value Checker annotations `@IntRange`.  This is probably the better approach
These will be trusted rather than checked.

Note that `array.length()` is special-cased by the Index Checker (there is nowhere to write an annotation), and so if you use Index Checker annotations, the Signedness Checker should probably special-case it too.  (The alternative would be to have the Signedness Checker run the Index Checker, but the Index Checker is very slow and so this seems to heavyweight to me.

**Comment by gokaco** [comment]
Consider the following code-
```
import org.checkerframework.checker.signedness.qual.*;
class ex {
   void foo(String s){
    	int b=s.length();
    	@Unsigned int c=b;
    }
}
```
It gives no errors. Here Value Checker worked. Does it shows that the Signedness Checker is working with Value Checker on variable ```b``` by determining its range to be within ```0 & Integer.MAX_VALUE``` but not on method directly?

**Comment by gokaco** [comment]
Fixed through #2614
